drivers/docker/driver.go

```diff
@@ -612,7 +612,15 @@ func (d *Driver) loadImage(task *drivers.TaskConfig, driverConfig *TaskConfig, c
 }
 
 func (d *Driver) containerBinds(task *drivers.TaskConfig, driverConfig *TaskConfig) ([]string, error) {
-
+	if runtime.GOOS == "windows" {
+		imageConfig, _ := client.InspectImage(driverConfig.Image)
+		// LCOW If we are running a Linux Container on Windows, we need to mount it correctly, as c:\ does not exist on unix
+		if imageConfig.OS == "linux" {
+			task.Env[taskenv.AllocDir] = strings.ReplaceAll(task.Env[taskenv.AllocDir], "c:\\", "/")
```

Comment:
Does this handle the Nomad datadir being on a non-`C:\` partition safely? And do we also need to replace the rest of the `\` in the path?

Reply:
As the Linux image does not have a c: unit, the `C:\alloc` directory must be `/alloc`. From my tests, Docker handles correctly the windows-to-linux mounts.

From my point of view, AllowDir, SecretsDir and LocalDir do not have more than one `\`, but it's true that can be changed. New replacement modification incoming.

Reply:
I have been thinking about this. SharedAllocName, TaskLocal and TaskSecrets are  hardcoded in alloc_dir.go, so they should never have anymore `\` in the path.

However, the latter modification handles all the possible paths, so it might be better use that.